Add Hyperledger Bevel hip

tkuhrt opened the pull request.

tkuhrt committed:
Add BAF Proposal

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add content

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add content

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Remove files

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add images

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Correct grammar

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add documentation and good first issue links

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Changed reference section to links

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add FAQ on related projects/labs

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

mwklein committed:
Updated description of tool usage within BAF

Signed-off-by: Michael Klein <michael.w.klein@accenture.com>

tkuhrt committed:
Merge pull request #1 from mwklein/add-baf-hip

Updated description of tool usage within BAF

tkuhrt committed:
Correct typos

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add high-level description for Ansible

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Update formatting

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Add sponsor

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

sownak committed:
Add new sponsor

Signed-off-by: Roy,Sownak <sownak.roy@accenture.com>

tkuhrt committed:
Merge pull request #2 from sownak/add-baf-hip

Add new sponsor

suvajit-sarkar committed:
Adding sponsor

Signed-off-by: suvajit sarkar <suvajit.sarkar@accenture.com>

tkuhrt committed:
Merge pull request #3 from suvajit-sarkar/add-baf-hip

Add new sponsor

suvajit-sarkar committed:
Added sponsor

Signed-off-by: suvajit sarkar <suvajit.sarkar@accenture.com>

tkuhrt committed:
Merge pull request #4 from suvajit-sarkar/add-baf-hip

Add new sponsor

arsulegai committed:
Add Arun as a sponsor

Signed-off-by: S m, Aruna <aruna.mohan@walmart.com>

tkuhrt committed:
Merge pull request #5 from arsulegai/add-arun

Add Arun as a sponsor

tkuhrt committed:
Merge branch 'hyperledger:gh-pages' into add-baf-hip

ryjones committed:
Revert "Update logo"

This reverts commit 5aa4010d66611b006721c9f1b850e080f0df566a.

Signed-off-by: Ry Jones <ry@linux.com>

tkuhrt committed:
Merge branch 'hyperledger:gh-pages' into add-baf-hip

tkuhrt committed:
Update title

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Added FAQ on incubation entry considerations

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Corrected typo

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Updated integration with fabric-operations-console lab.

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt committed:
Formatting

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

jimthematrix commented on HIPs/baf.md:
```diff
@@ -0,0 +1,248 @@
+
+Therefore, when blockchain developers are asked to use an unfamiliar DLT/Blockchain platform, it requires significant effort for even experienced technicians to properly setup the DLT/Blockchain network. This is especially true in large-scale production projects across heterogeneous corporate environments which require other key aspects such as security and service availability.
+
+Being aware of the potential difficulty and complexity of getting a production-scale DLT/Blockchain network ready, cloud vendors have provisioned their own managed Blockchain services (aka Blockchain as a Service or BaaS) to help alleviate various pain-points during the process. However, limitations can still be identified in these BaaS solutions (e.g., limited network size, locked to all nodes o ...
```
I think the motivation for BAF should not be painted as "an alternative to BaaS" or "addressing limitations of BaaS",  because BAF vs. BaaS are apples and oranges.

BAF certainly takes care of the provisioning of blockchain networks, but that's only a (small) part of the blockchain network lifecycle management, other aspects include:
- consortium governance
- ongoing software upgrades while the network is running, including complex orchestration procedures like hard forks (such as for Ethereum based networks)
- monitoring metrics (both low-level resources metrics as well as DLT-level consensus liveness checks)
- backup and recovery
- etc.

High quality BaaS platforms offer all the above, plus the support that nodes in the same blockchain network can be run either in the cloud or on-prem (including private data centers and dedicated cloud accounts).

It may be OK to assume that one or a few parties in a consortium are capable of provisioning and operating blockchain nodes themselves, it'd be a mistake to assume that all parties are capable of and willing to doing that. As the saying goes "you are as fast as your slowest party", BaaS based fully managed nodes must be considered as part of any enterprise consortium's deployment options.

I think it would be more accurate to say that BAF is complementary to BaaS. It'd be interesting to see what it would take to make BAF work together with some leading BaaS platforms so it can help deploy on-prem nodes.

Finally, some BaaS platforms are not subject to the cited limitations (limited network size, locked to all nodes on a single cloud provider, limited choice of DLT/Blockchain platform) at all. So that sentence should be clarified.

tkuhrt replied:
Agree that the limitations are not inherent in all BaaS. But these reasons are the motivations for creating BAF. We would love to see some BaaS providers use BAF in their solutions.

jimthematrix commented on HIPs/baf.md:
```diff
@@ -0,0 +1,248 @@
+* Bring Your Own Infrastructure (BYOI) - You provide GIT, Kubernetes cluster(s), and Hashicorp Vault services provisioned to meet your specific requirements and enterprise standards
+* No network size limit
+* Specifies only the number of organizations and the number of nodes per organization in a network.yaml file uniquely designed in the Blockchain Automation Framework for a new DLT/Blockchain network set-up and its future maintenance
+* Provides a sample supply chain application which runs on multiple DLT/Blockchain platforms that can be used as a reference pattern for how to safely abstract application logic from the underlying DLT/Blockchain platform
```
Can we add the list of aspects inherent to BaaS but are not covered in BAF?
- consortium governance
- ongoing software upgrades while the network is running, including complex orchestration procedures like hard forks (such as for Ethereum based networks)
- monitoring metrics (both low-level resources metrics as well as DLT-level consensus liveness checks)
- backup and recovery
- 24/7 support

tkuhrt replied:
I think it is more interesting to consider how BAF differs from BaaS https://github.com/hyperledger/hyperledger-hip/blob/0e70c4b6ae99cd8d788a1110162552a8eff58471/HIPs/baf.md#how-is-blockchain-automation-framework-different-from-blockchain-as-a-service-baas

tkuhrt replied:
My bad. This is that section. Should not review on a phone. We could add that here.

sownak replied:
- Monitoring metrics (using prometheus or splunk)  is in the roadmap.
- Software upgrades like upgrade from Fabric 1.4.3 to 2.2.2, will be added in the roadmap as it has been planned as a feature
- Backup and recovery is provided by the underlying Infrastructure as we are saying Bring-your-own-infra

jimthematrix replied:
> Software upgrades like upgrade from Fabric 1.4.3 to 2.2.2, will be added in the roadmap as it has been planned as a feature

this one was not about support for a specific ledger version, but to support seamless upgrade from one version to another at the user's discretion. Using an example: say I have used BAF to provision an Ethereum network with Besu version X, and would like to upgrade to version Y which supports the London hard fork, the following must happen for a smooth upgrade:
- all node owners agree on a future block number for the hardfork to become effective
- all nodes configurations get re-initialized with an updated genesis block that specifies the agreed block number for the London hardfork EIP
- all nodes must be upgraded before the chain height gets to the agreed block number
- the upgrade process be conducted in a rolling fashion to allow the network to continue to serve client applications during the upgrade

jimthematrix replied:
@sownak thanks for clarifying. Glad to see that monitoring is in the roadmap 👍 . I agree that backup and recovery is covered under the BYOI assumption

ryjones commented:
@tkuhrt @sownak : do you want to wait until BAF has a new name to merge? I don't think it is a requirement as long as it gets fixed later

tkuhrt commented:
Approved [November 4, 2021 TSC Meeting](https://wiki.hyperledger.org/display/TSC/2021+11+04+TSC+Meeting+Record).

tkuhrt commented:
> @tkuhrt @sownak : do you want to wait until BAF has a new name to merge? I don't think it is a requirement as long as it gets fixed later

I am okay if it gets merged now and a new PR created for the rename of the file with updates for the name.

tkuhrt committed:
Renamed BAF to Hyperledger Bevel

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt commented:
@ryjones : I renamed the HIP to use Hyperledger Bevel. I also went through and changed Blockchain Automation Framework/BAF to Hyperledger Bevel in the text. There are still some references to Blockchain Automation Framework where it made sense to reference its name as a lab. We should be able to get this merged now.

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+Hyperledger Bevel (formerly Blockchain Automation Framework) HIP v0.1
+
+# Sponsor(s)
+<mark>_**Sponsor(s)** name and contact details including email address_</mark>
```
elide?

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+- Arun S M, Staff Software Engineer, Walmart Global Tech India, TSC member (aruna.mohan@walmart.com)
+
+# Abstract
+<mark>_**Abstract** (less than 50 word) description of the project._</mark>
```
same as above

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+The Hyperledger Bevel is an accelerator by which developers can consistently deploy production-ready distributed networks across public and private cloud providers.
+
+# Context
+<mark>_**Context** if any, what is this project derived from? What if any is it related to?_</mark>
```
same as above

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+In the [Hyperledger Labs environment](https://github.com/hyperledger-labs/blockchain-automation-framework), we have been working with others in the community and performing a number of [releases](https://github.com/hyperledger-labs/blockchain-automation-framework/releases). We have a [chat channel](https://chat.hyperledger.org/channel/blockchain-automation-framework), a [wiki page](https://wiki.h ...
+
+# Dependent Projects
+<mark>_**Dependent Projects** if any, must be listed, and each dependent project\'s maintainers must sign off on the proposal before it is considered by the TSC._</mark>
```
same as above

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+* Corda Enterprise
+
+# Motivation
+<mark>_**Motivation** for this project, a longer justification of the project may be a couple of hundred words. Why is this project better at solving a problem compared to parallel proposals or implemented projects?_</mark>
```
again

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+The objective of Hyperledger Bevel is to provide a consistent means by which developers can deploy production-ready distributed networks across public and private cloud providers. This enables developers to focus on building business applications, knowing that the framework upon which they are building can be adopted by an enterprise IT production operations organization. Hyperledger Bevel is not ...
+
+# Proposed Status
+<mark>_**Status** of the project: See [project lifecycle](https://hyperledger.github.io/tsc/project-lifecycle.html)._</mark>
```
again

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+Incubation
+
+# Solution
+<mark>_**Solution** to the problem addressed in the motivation section. Try to make this as detailed as possible._</mark>
```
again

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+![Hyperledger Bevel for Corda Open Source](./images/bevel/bevel-corda.png "Hyperledger Bevel for Corda Open Source")
+
+# Effort and Resources
+<mark>_**Effort and resources** committed (coders and any other resources that are needed) and timeline._</mark>
```
again

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+
+
+# Closure
+<mark>_**Closure** how do we know that the project succeeded. This has to be measurable if possible. Make references to successor projects if any._</mark>
```
again

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+The Hyperledger Bevel documentation provides detailed [Operation Guide](https://blockchain-automation-framework.readthedocs.io/en/latest/operationalguide.html) and [Developer Guide](https://blockchain-automation-framework.readthedocs.io/en/latest/developerguide.html). In addition, a [supply chain sample application](https://blockchain-automation-framework.readthedocs.io/en/latest/example/supplych ...
+
+# References
+<mark>_**References**. See [citation guide](http://www.chicagomanualofstyle.org/tools_citationguide.html)._</mark>
```
again

ryjones commented on HIPs/bevel.md:
```diff
@@ -0,0 +1,248 @@
+Since being accepted as a Hyperledger Lab in October 2019, there have been 41 total contributors with 35 contributors having more than one commit (based on metrics obtained from the [Hyperledger community tools project reports](https://github.com/tkuhrt/hyperledger-community-management-tools/tree/master/project-reports) on October 27, 2021). There have been 26 contributors in the past year and 14 ...
+
+# How To
+<mark>_**How to**: How to host and test the project. How to deploy and use. How does one know that it works._</mark>
```
again

ryjones reviewed (commented):
Hi @tkuhrt - I propose deleting the marked lines; I don't feel strongly about it; I looked at the Firefly HIP, and they were deleted from that.

tkuhrt committed:
Removed mark lines

Signed-off-by: Tracy Kuhrt <tracy.a.kuhrt@accenture.com>

tkuhrt commented:
> Hi @tkuhrt - I propose deleting the marked lines; I don't feel strongly about it; I looked at the Firefly HIP, and they were deleted from that.

Done
